src/FileDropper/FileDropper.tsx

```diff
@@ -0,0 +1,193 @@
+import { CloudUploadIcon } from 'lucide-react';
+
+import {
+  Alert,
+  Box,
+  LinearProgress,
+  Stack,
+  SxProps,
+  Typography,
+  useTheme,
+} from '@mui/material';
+
+import { DndProvider, DropTargetMonitor, useDrop } from 'react-dnd';
+import { HTML5Backend, NativeTypes } from 'react-dnd-html5-backend';
+
+import UploadFileButton, {
+  UploadFileButtonProps,
+} from '../buttons/UploadFileButton/UploadFileButton';
+
+export type FileDropperProps = {
+  id?: string;
+  sx?: SxProps;
+  /**
+   * Callback on files dropped
+   */
+  onDrop?: (files: File[]) => void;
+  /**
+   * Callback on files selected
+   */
+  onChange: UploadFileButtonProps['onChange'];
+  /**
+   * Error to show
+   */
+  error?: string;
+  /**
+   * Smaller text to show, such as limits
+   */
+  hints?: string;
+  /**
+   * Text of the browsing button
+   */
+  buttonText?: string;
+  /**
+   * additional buttons to display next to browse files
+   */
+  buttons?: JSX.Element;
+  /**
+   * When true, allows many files to be selected when browsing
+   */
+  multiple?: UploadFileButtonProps['multiple'];
+  /**
+   * Text displayed by default
+   */
+  message?: string;
+  /**
+   * Text displayed on drag enter
+   */
+  releaseText?: string;
+  /**
+   * Whether files are getting uploaded
+   */
+  isLoading?: boolean;
+  /**
+   * If `isLoading` is true, shows given `upload progress` value in a progress bar
+   */
+  uploadProgress?: number;
+};
+
+const FileDropperComponent = ({
+  id,
+  onDrop,
+  onChange,
+  error,
+  hints,
+  sx,
+  buttonText = 'Browse files',
+  buttons,
+  multiple,
+  message = `Drag your files here to upload or`,
+  releaseText = 'Release to drop',
+  uploadProgress = 0,
+  isLoading = false,
+}: FileDropperProps): JSX.Element | null => {
+  const theme = useTheme();
+
+  const [{ canDrop, isOver }, drop] = useDrop({
+    accept: [NativeTypes.FILE],
+    drop: (e: { files: File[] }) => {
+      onDrop?.(e.files);
+    },
+    canDrop: () => {
+      return !isLoading;
+    },
+    collect: (monitor: DropTargetMonitor) => ({
+      isOver: monitor.isOver(),
+      canDrop: monitor.canDrop(),
+    }),
+  });
+
+  const isActive = canDrop && isOver;
+  let bgColor = '#eeeefa';
+  if (isActive) {
+    bgColor = '#dcdcf6';
+  } else if (!canDrop && isOver) {
+    bgColor = '#ffbaba';
+  }
+
+  return (
+    <Stack
+      role='dropzone'
+      id={id}
+      height='100%'
+      width='100%'
+      display='flex'
+      justifyContent='center'
+      alignItems='center'
+      bgcolor={error ? '#ffbaba' : bgColor}
+      borderRadius={5}
+      ref={drop}
+      gap={2}
+      sx={{
+        borderStyle: 'dashed',
+        borderWidth: 3,
+        borderColor: error ? 'red' : 'lightgrey',
+        ...sx,
```

Comment:
Is it really necessary to explose a `sx` ? It feels like we just do not know how it should be re-used, so we stuff a "put everything you want" prop to account for that.